services/web/client/source/class/qxapp/component/filter/IFiltrable.js

```diff
@@ -0,0 +1,83 @@
+/* ************************************************************************
+
+   qxapp - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2019 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Ignacio Pascual (ignapas)
+
+************************************************************************ */
+
+/**
+ * Defines the methods all UI filtrable element should implement.
+ */
+qx.Interface.define("qxapp.component.filter.IFiltrable", {
+  members:{
+    /**
+     * Function in charge of filtering the element out of the interface.
+     * It will usually hide it or decrease its opacity, but it could also trigger some other UI changes.
+     *
+     * @param {Object?} data The data contained in the message.
+     * @param {qx.event.message.Message?} msg Incoming message containing the data.
+     */
+    _filterOut: function(data, msg) {
+      this.assertArgumentsCount(arguments, 0, 2);
+      if (msg) {
+        this.assertInstance(msg, qx.event.message.Message);
+      }
+    },
+
+    /**
+     * Function in charge of taking the filter out of the element.
+     * It will usually make it visible again, but it could also trigger some other UI changes.
+     *
+     * @param {Object?} data The data contained in the message.
+     * @param {qx.event.message.Message?} msg Incoming message containing the data.
+     */
+    _unfilter: function(data, msg) {
```

Comment:
I would make sure the two function names are properly related: eg `_setFilteredLook` - `_unsetFilteredLook`